include/swift/Localization/LocalizationFormat.h

```diff
@@ -154,14 +155,22 @@ class SerializedLocalizationWriter {
 };
 
 class LocalizationProducer {
+  /// This allocator will retain localized diagnostic strings containing the
+  /// diagnostic's message and identifier for the duration of compiler
+  /// invocation
```

Comment:
I think it's worth explaining when exactly this is going to be used.

Reply:
WDYT about 
```
  /// This allocator will retain localized diagnostic strings containing the
  /// diagnostic's message and identifier as `message [id]` for the duration of
  /// compiler invocation. This will be used when the frontend flag
  /// `-debug-diagnostic-names` is used.
```
I tried to keep it short, yet helpful.

Reply:
Sounds good!